File: backend/internal/models/payment.go
package models

import "time"

type PaymentStatus string

const (
	PaymentStatusPending PaymentStatus = "pending"
	PaymentStatusSuccess PaymentStatus = "success"
	PaymentStatusFailed  PaymentStatus = "failed"
)

type Payment struct {
	ID        string        `json:"id"`
	RentalID  string        `json:"rental_id"`
	UserID    string        `json:"user_id"`
	Amount    int64         `json:"amount"`
	Provider  string        `json:"provider"`
	Status    PaymentStatus `json:"status"`
	CreatedAt time.Time     `json:"created_at"`
}

File: frontend/src/app/core/models/payment.ts
export type PaymentStatus = 'pending' | 'success' | 'failed';

export interface Payment {
  id: string;
  rental_id: string;
  user_id: string;

  // From DTO Join
  user_name: string;
  storage_number: string;

  amount: number;
  provider: string; // 'kaspi', 'card', etc.
  status: PaymentStatus;

  created_at: string;
}
